src/gwt/src/org/rstudio/core/client/widget/DataTableColumnWidget.java

```diff
@@ -0,0 +1,92 @@
+package org.rstudio.core.client.widget;
+
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.event.dom.client.*;
+import org.rstudio.core.client.CommandWith2Args;
+import org.rstudio.core.client.Debug;
+import org.rstudio.core.client.theme.res.ThemeResources;
+import org.rstudio.core.client.theme.res.ThemeStyles;
+
+import java.util.function.BiConsumer;
+
+public class DataTableColumnWidget extends TextBox
+{
+   public DataTableColumnWidget(BiConsumer<Integer, Integer> onEnter)
+   {
+      onEnterFunction_ = onEnter;
+      ThemeStyles styles = ThemeResources.INSTANCE.themeStyles();
+      setStylePrimaryName(styles.dataTableColumnWidget());
+      getElement().setAttribute("spellcheck", "false");
+
+      init();
+   }
+
+   /**
+    * Parse the contents of getValue() and return an array with offset and max
+    * Valid values of the textbox are either a single integer or two integers separated by a dash, spaces are trimmed
+    * Any other values of the textbox are ignored
+    * @return int[offset, max]
+    */
+   private int[] getValueOffsetAndMax()
+   {
+      int[] offsetAndMax = {-1, -1};
+
+      String value = getValue();
+      if (value == null || value.length() < 1) { return offsetAndMax; }
+
+      String[] values = value.split("-");
+      try
+      {
+         // starting at column 1 means no offset
+         offsetAndMax[0] = Integer.parseInt(values[0].trim()) - 1;
+      } catch (NumberFormatException e)
```

Comment:
Can we keep `}` and `catch()` on separate lines? At least in Java sources we almost always keep `{` and `}` on their own lines (with exceptions for e.g. lambdas)

Reply:
Yeah this is all a matter of me configuring IntelliJ's auto formatter, I must have missed this one as all of the defaults have things on the same line. It's also compounded by going between Javascript and Java using differing styles.